storage/remote/client.go

```diff
@@ -169,6 +179,10 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	if err != nil {
 		return nil, errors.Wrap(err, "error sending request")
 	}
+	if span != nil {
+		span.Finish()
```

Comment:
already done in L162

Reply:
This is to close the child span just after the http request